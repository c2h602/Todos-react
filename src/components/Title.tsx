import { memo } from "react";

const Title = memo(({ text }: { text: string }) => {
  return <h1>{text}</h1>;
});

export default Title;