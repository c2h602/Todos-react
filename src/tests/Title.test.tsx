import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Title from '../components/Title';

describe('Title', () => {
    it(`корректно рендерит заголовок 'todos'`, () => {
        const { container } = render(<Title text='todos'/>);
        expect(container).toMatchSnapshot();
        
        const elem = screen.getByText('todos');
        expect(elem).toBeInTheDocument();
    });
});
