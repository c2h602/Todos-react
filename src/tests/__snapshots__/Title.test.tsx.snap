// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title корректно рендерит заголовок 'todos' 1`] = `
<div>
  <h1>
    todos
  </h1>
</div>
`;
